src/main/java/by/custom/utilcalculator/controller/MessageRouter.java

```diff
@@ -35,28 +35,30 @@ public SendMessage route(Update update) {
     //we receive message from user, check it and handle it if it is ok here
     private SendMessage getCheckInputMessageAndGetAnswer(Message message) {
         String usersMessage = message.getText();
-        String userID = message.getChatId().toString();
-        UserProgress userProgress = UserStorageManager.getInstance().getUserProgress(userID);
+        String chatID = message.getChatId().toString();
         String answer;
 
         switch (usersMessage) {
-            case Command.START -> answer = getGreetingMessage();
+            case Command.START -> {
+                userProgressManager.createNewUserProgress(chatID);
+                answer = getGreetingMessage();
+            }
             case Command.EAES, Command.OTHER_COUNTRIES ->
-                    answer = botFieldsManager.processCarOrigin(usersMessage, userProgress);
+                    answer = userProgressManager.processCarOrigin(usersMessage, chatID);
             case Command.PHYSICAL_PERSON, Command.JURIDICAL_PERSON ->
-                    answer = botFieldsManager.processOwnerType(usersMessage, userProgress);
+                    answer = userProgressManager.processOwnerType(usersMessage, chatID);
             case Command.LESS_3_YEARS_AGE, Command.BETWEEN_3_AND_7_YEARS_AGE, Command.MORE_7_YEARS_AGE ->
-                    answer = botFieldsManager.processCarAge(usersMessage, userProgress);
+                    answer = userProgressManager.processCarAge(usersMessage, chatID);
             case Command.GASOLINE_TYPE_ENGINE, Command.ELECTRIC_TYPE_ENGINE ->
-                    answer = botFieldsManager.processEngineType(usersMessage, userProgress);
+                    answer = userProgressManager.processEngineType(usersMessage, chatID);
             case Command.VOLUME_LESS_1000_CM, Command.VOLUME_BETWEEN_1000_2000_CM, Command.VOLUME_BETWEEN_2000_3000_CM,
                     Command.VOLUME_BETWEEN_3000_3500_CM, Command.VOLUME_MORE_3500_CM ->
-                    answer = botFieldsManager.processEngineVolume(usersMessage, userProgress);
+                    answer = userProgressManager.processEngineVolume(usersMessage, chatID);
             default -> answer = getSorryMessage();
         }
 
         SendMessage sendMessage = new SendMessage();
```

Comment:
I recommend to use appropriate constructor and remove default(without params) constructor. In this case, it's impossible to create "invalid" object, an object with `null` fields. 
```suggestion
        SendMessage sendMessage = new SendMessage(chatId, answer);
```

Reply:
The brilliant idea, i like it. Done.